### PR TITLE
code-first-backend-api

### PR DESCRIPTION
**Pull request by lawrence415610**
Accomplish all the requirements on the grading list

### CODE REVIEW

File: backend/src/controller/user.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        password,
+      },
+    });
+    const userWithoutPassword = { ...newUser, password: undefined };
+    return res.status(201).json({
+      success: true,
+      data: userWithoutPassword,
+    });
+  } catch (error) {
+    console.log(error);
```
Comment (MarioBanovac) [inline backend/src/controller/user.ts:72]: We can remove this console.log

File: backend/src/controller/user.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        success: false,
+        error: Error.UserNotFound,
+      });
+    const { password, ...userWithoutPassword } = user;
+    res.status(200).json({
+      success: true,
+      data: userWithoutPassword,
+    });
+  } catch (error) {
+    console.log(error);
```
Comment (MarioBanovac) [inline backend/src/controller/user.ts:99]: Also here and on every catch method :)

File: backend/src/controller/user.ts [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        error: Error.UsernameAlreadyTaken,
+        success: false,
+      });
+    }
+
+    const updatedUser = await User.update({
+      where: {
+        id: Number(id),
+      },
+      data: { ...user!, ...req.body },
```
Comment (MarioBanovac) [inline backend/src/controller/user.ts:154]: What is the meaning of the exclamation mark?

**Review by MarioBanovac — COMMENTED** [review]
I took a look at your code and it looks great! Keep up the good work

File: backend/src/controller/user.ts (hunk 3 above)
Comment (lawrence415610) [reply backend/src/controller/user.ts:154]: This is to suppress the warning of typescript, the user don't have a chance to be null

Comment (MarioBanovac) [reply backend/src/controller/user.ts:154]: Ok, sounds good

**Pull request merged**